frontend/app/services/MemberService.scala

```diff
@@ -110,73 +111,85 @@ class MemberService(identityService: IdentityService,
+    def createStripeCustomer(stripeToken: String): Future[Customer] =
+      stripeService.Customer.create(user.id, stripeToken).andThen {
+        case Failure(e) => logger.warn(s"Could not create Stripe customer for user ${user.id}", e)
+      }
 
     def updateIdentity(): Future[Unit] =
       Future {
         formData.password.foreach(identityService.updateUserPassword(_, identityRequest, user.id)) // Update user password (social signin)
         identityService.updateUserFieldsBasedOnJoining(user, formData, identityRequest) // Update Identity user details in MongoDB
-      }.andThen { case Failure(e) => logger.error(s"Could not update Identity for user ${user.id}", e)}
+      }.andThen { case Failure(e) => logger.error(s"Could not update Identity for user ${user.id}", e) }
 
-    def createPaidZuoraSubscription(sfContact: ContactId, stripeCustomer: Customer, paid: PaidMemberJoinForm, email: String): Future[String] =
+    def createPaidZuoraSubscription(sfContact: ContactId, paid: PaidMemberJoinForm, email: String, stripeCustomer: Option[Customer]): Future[String] =
       (for {
         zuoraSub <- createPaidSubscription(sfContact, paid, paid.name, paid.tier, stripeCustomer, campaignCode, email, ipAddress, ipCountry)
       } yield zuoraSub.subscriptionName).andThen {
-        case Failure(e: PaymentGatewayError) => logger.warn(s"Could not create paid Zuora subscription due to payment gateway failure: ID=${user.id}; Stripe Customer=${stripeCustomer.id}", e)
-        case Failure(e) => logger.error(s"Could not create paid Zuora subscription: ID=${user.id}; Stripe Customer=${stripeCustomer.id}", e)
+        case Failure(e: PaymentGatewayError) => logger.warn(s"Could not create paid Zuora subscription due to payment gateway failure: ID=${user.id}", e)
+        case Failure(e) => logger.error(s"Could not create paid Zuora subscription: ID=${user.id}", e)
       }
 
     def createFreeZuoraSubscription(sfContact: ContactId, formData: JoinForm, email: String) =
       (for {
         zuoraSub <- createFreeSubscription(sfContact, formData, email, ipAddress, ipCountry)
       } yield zuoraSub.subscriptionName).andThen { case Failure(e) =>
-        logger.error(s"Could not create free Zuora subscription for user ${user.id}", e)}
+        logger.error(s"Could not create free Zuora subscription for user ${user.id}", e)
+      }
 
     def updateSalesforceContactWithMembership(stripeCustomer: Option[Customer]): Future[ContactId] =
       salesforceService.updateMemberStatus(user, tier, stripeCustomer).andThen { case Failure(e) =>
-        logger.error(s"Could not update Salesforce contact with membership status for user ${user.id}", e)}
+        logger.error(s"Could not update Salesforce contact with membership status for user ${user.id}", e)
+      }
 
     formData match {
-      case paid: PaidMemberJoinForm => //
+      case paid@PaidMemberJoinForm(_, _, PaymentForm(_, _, Some(_)), _, _, _, _, _, _, _, _, _) => //Paid member with PayPal token
```

Comment:
Yup, you're right, seems to only work as varargs in sequences. Sorry, ignore this comment then 😬.